src/atom.ts

```diff
@@ -120,6 +127,24 @@ function mkAtom<A>(observable: B.Property<A>, get: () => A, modify: ( (fn: (a :
             return lensedAtom(theAtom, lens)
         }        
     }
+    theAtom.case = (...fns: any) => {
+        const matchingBranch = (value: A) => {
+            let i = 0
+            while (i < fns.length) {
+                const match = fns[i++]
+                const map = fns[i++]
+                if (match(value)) return { i, match, map }
+            }
+            throw new Error(`No branch matched value ${value}`)
+        }
+
+        return theAtom.skipDuplicates((left, right) => {
+            return matchingBranch(left).i === matchingBranch(right).i
+        }).flatMapLatest(value => {
+            const { match, map } = matchingBranch(value)
+            return map(theAtom /* .freezeUnless(match) */)
```

Comment:
Uncommenting `.freezeUnless(match)` here breaks it, as it thinks `theAtom` has no value currently. I think that `.freezeUnless` is required to avoid rendering once more after the discriminated union branch changes.

Reply:
Seems to work with latest changes :+1: